Amaan taking over for Akshay - feat: Implement dark mode for Consumables/Materials page with full-pa…

vamsidharpanithi opened the pull request:
# Description

**Fixes:** (PRIORITY MEDIUM) Jae/Sudarshan – Fix Dark Mode Functionality for the Consumables Page
The Consumables page did not update when switching to dark mode. The page remained in light mode while other dashboard components correctly responded to theme changes. This PR implements full dark mode support for the entire Materials/Consumables section.


## Related PRS (if any):
This frontend PR is related to the backend  [1262](https://github.com/OneCommunityGlobal/HGNRest/pull/1262) (HGNRest).

…

## Main changes explained:
- Modified ItemListView.jsx: Added darkMode state, useEffect for #root background, conditional classes
- Created ItemListView.module.css: Converted to CSS module with comprehensive dark mode styles
- Modified ItemsTable.jsx: Added darkMode prop and table styling
- Modified SelectForm.jsx & SelectItem.jsx: Added darkMode prop and input styling
- Modified UpdateConsumable.jsx & UpdateConsumableModal.jsx: Added darkMode support
- Created UpdateConsumable.module.css: Converted to CSS module with dark mode styles
- Deleted old CSS files: Converted to modules
- Modified package.json: Added OpenSSL legacy provider support for Node.js 20
- Created scripts/start-with-openssl.js: Wrapper script for OpenSSL compatibility

…

## How to test:
1. Check out branch: vamsidhar-consumables-dark-mode
2. Run npm install
3. Run npm start
4. Navigate to bmdashboard/consumables
5. Toggle dark mode and verify all elements adapt correctly
6. Test both Consumables and Materials pages

## Screenshots or videos of changes:
<img width="1676" height="851" alt="image" src="https://github.com/user-attachments/assets/90f5de95-7950-4936-b7ca-1979f1b50c0d" />

<img width="1680" height="848" alt="image" src="https://github.com/user-attachments/assets/7c7fce95-2357-459b-b31e-3b9ad6a22741" />

vamsidharpanithi committed:
feat: Implement dark mode for Consumables/Materials page with full-page dark background and styling for all UI elements

vamsidharpanithi committed:
Merge development branch and resolve conflicts - keep dark mode changes

vamsidharpanithi committed:
Resolve package.json merge conflict

vamsidharpanithi committed:
Update pre-push hook to check for dependencies

vamsidharpanithi committed:
Merge development branch and resolve conflicts - keep dark mode changes

vamsidharpanithi committed:
Fix redux-mock-store import: change from named to default import in 87 test files

vamsidharpanithi committed:
Fix linting errors and allow App.css in pre-commit hook

Anusha-Gali reviewed (changes requested):
Hi Vamsidhar,

I tried to review your PR locally but it fails to open the page with many errors, below is a glimpse of few of the errors.
<img width="1278" height="756" alt="Screenshot 2026-01-15 at 12 44 23 AM" src="https://github.com/user-attachments/assets/8b1716f8-bb14-4eea-b1fa-bc812841c920" />
<img width="1278" height="756" alt="Screenshot 2026-01-15 at 12 44 27 AM" src="https://github.com/user-attachments/assets/563179c7-706c-4467-acea-c7c5a27a40b2" />

kzou55 reviewed (changes requested):
Hi Saurabh, 

I tried to checkout and test the current branch. However, whenever I do `npm install`, I get the following errors.

  <img width="625" height="495" alt="Image" src="https://github.com/user-attachments/assets/8e24cf2d-3884-498a-a926-83959ee3f127" />

Usually the following would work:
- `Remove-Item -Recurse -Force node_modules` 
- `Remove-Item package-lock.json`

And then another `npm install` but this gets an error too. 

<img width="476" height="32" alt="Image" src="https://github.com/user-attachments/assets/4c055766-f2af-42da-8529-c2a1e76ad431" />

saurabhdipte committed:
Merge origin/development into vamsidhar-consumables-dark-mode - resolve conflicts

saurabhdipte committed:
Update yarn.lock after merge

saurabhdipte committed:
Fix SonarCloud issues: remove duplicate case-variant files, fix start-with-openssl.js

saurabhdipte committed:
Fix start-with-openssl.js optional chain logic bug

saurabhdipte committed:
Fix start-with-openssl.js to use optional chaining

saurabhdipte committed:
Refactor UpdateConsumable to reduce cognitive complexity

saurabhdipte committed:
Refactor UpdateConsumable to reduce cognitive complexity

saurabhdipte committed:
Fix UpdateConsumable submitHandler syntax error

saurabhdipte committed:
Fix UpdateConsumable - remove duplicate code, clean up file

saurabhdipte committed:
Fix SonarCloud negated condition warning in UpdateConsumable

saurabhdipte committed:
Fix prettier formatting in UpdateConsumable.jsx

saurabhdipte committed:
Extract duplicate onKeyDown/onInput handlers to reduce duplication

saurabhdipte committed:
Fix prettier formatting in start-with-openssl.js

DeepighaJ reviewed (changes requested):
Test the PR locally and the page loads goods in the dark mode but there are few minor issues
- Add Material, Edit Name/Measures and View Update History buttons are not clickable.
- The Use record View pop up is empty and the header need contrast improvement.
- Update Record and Purchases View popup title also needs contrast improvement

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/5ee62a89-1ab4-4978-ae6f-08ef901a7bbb" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/4a869bb7-0848-470b-bbec-11fa5af9be9d" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/83f83b5b-e751-4350-92d5-f1ab95fd884a" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/1b275431-6641-4660-9d6a-8fbe2426bf4e" />

kzou55 reviewed (changes requested):
Hi Saurabh, 

I ran the PR locally again. 

Still getting a lot of import issues. 

<img width="1693" height="685" alt="Image" src="https://github.com/user-attachments/assets/0d85e49f-6423-498d-be0d-b2b9b747d81d" />

<img width="1310" height="896" alt="Image" src="https://github.com/user-attachments/assets/b9c33a1c-701e-4457-b00d-909f933bd556" />

kzou55 reviewed (changes requested):
## Verified
- The following look fine in dark mode
  - Date Component 
  - Project & Consumables dropdown
  - Search component
  - The Table

  <img width="1609" height="938" alt="Image" src="https://github.com/user-attachments/assets/c508bf36-facc-4e50-a1f0-ae71de86f909" />


## Issues
1.  Dark Mode expands the bottom of the page a bit making it inconsistent with the layout in Light Mode. Notice how in Dark Mode the bottom of the page is a bit expanded with blank spaces

    <img width="1917" height="1036" alt="Image" src="https://github.com/user-attachments/assets/18ed3a9e-304e-4113-800f-b01a66433881" />

    <img width="1912" height="1031" alt="Image" src="https://github.com/user-attachments/assets/25dae7d7-5c4f-45e8-8347-5a5545467cba" />

2. Poor header title in Dark Mode(Viewing Usage Record, Update Consumable Form, Purchase Record has poor text contrast )
    <img width="1252" height="959" alt="Image" src="https://github.com/user-attachments/assets/68a61c32-a00a-471e-b613-caa6b139cefe" />

    <img width="563" height="944" alt="Image" src="https://github.com/user-attachments/assets/633426c5-8f77-4c48-b763-bdc87c5a85ca" />

    <img width="1191" height="931" alt="Image" src="https://github.com/user-attachments/assets/b1ad1567-5920-4f84-a92d-e3e21f9e4aa3" />

## Note
- Assuming the following are unrelated to the PR/Out of Scope
  - Date Picker component has layout issues

  <img width="729" height="422" alt="Image" src="https://github.com/user-attachments/assets/ea4dba9f-7c82-4773-87a5-1ac6a68de03b" />
  - Add Material, Edit Name/Measurement, and View Update History button is not functional

  <img width="621" height="99" alt="Image" src="https://github.com/user-attachments/assets/85e6f054-f91f-438b-9111-e001529d58a0" />

sawantgayatri19 reviewed (changes requested):
Hi team, 

Thank you for working on the dark mode implementation for the Consumables and Materials page. After testing the current state, several critical bugs and styling issues need to be resolved before this can be approved:

1. Resolve Merge Conflicts:
   - There are active merge conflicts in `ItemListView.module.css`, `PermissionChangeLogTable.module.css`, and `yarn.lock`. Please rebase against `development` and clean these up.

2. Fix Functional Buttons:
   - The `Add Material`, `Edit Name/Measures`, and `View Update History` buttons are currently non-functional/un-clickable. Please verify their click handlers and event bindings.

3. Fix Empty Popups & Data Rendering:
   - The `Use Record View` popup renders completely empty. Ensure state is correctly passed down when opening modals.

4. Improve Dark Mode UI & Text Contrast:  

- Text Contrast:Fix low-contrast header text in dark mode for modals like 'Viewing Usage Record', 'Update Consumable Form', and 'Purchase Record'.

   - Layout Spacing: Address the extra blank space expanding the bottom of the page in dark mode to maintain layout parity with light mode.

Please push updates addressing these points so we can proceed with final testing!

akv-iu committed:
Merge origin/development into vamsidhar-consumables-dark-mode to sync before Consumables dark-mode fix

# Conflicts:
#	src/components/BMDashboard/ItemList/ItemListView.module.css
#	src/components/PermissionsManagement/PermissionChangeLogTable.module.css
#	yarn.lock

akv-iu committed:
refactor: update form group classes for consistent styling in ItemListView and Select components

akv-iu committed:
restore: revert unrelated files to development

An earlier conflict resolution (merge af8c6656f) kept this branch's side
wholesale, silently dropping ~105 lines of dark-mode CSS that development
had added in Header, Reports, TotalOrgSummary, UserProfile and
PermissionsManagement. Git recorded that as an intentional change on this
branch, so later merges could not restore it.

Restore all nine files to match origin/development so this PR contains only
Consumables work. Header/index.module.css additionally reorders one selector
list (no behaviour change) to satisfy stylelint no-descending-specificity.

akv-iu committed:
fix: sync package.json with lockfile to unblock CI

The Pull Request Unit Test job failed at 'yarn install --frozen-lockfile'.
package.json declared two devDependencies (jest-environment-jsdom-sixteen,
node-fetch) that yarn.lock had no entries for, so yarn refused the install.

Both packages are unused in this repo (no imports, and the test runner is
vitest, not jest) and neither exists on development, so remove them rather
than regenerate the lockfile. Also restores the trailing newline; package.json
is now identical to development.

Verified locally: yarn install --frozen-lockfile, yarn lint (0 errors) and
vitest --changed all pass (2149 tests).

kzou55 reviewed (changes requested):
Hi Akshay, 

I ran and tested the PR locally again. 

## Verified
- Consumables page

  <img width="1733" height="922" alt="Image" src="https://github.com/user-attachments/assets/2d3cfa50-58ab-46ee-8195-f1a8758a317f" />

  <img width="1727" height="990" alt="Image" src="https://github.com/user-attachments/assets/4888a7d8-bb26-40d6-b9f3-4f80c294902d" />
    -  View Usage Record

    <img width="1238" height="946" alt="Image" src="https://github.com/user-attachments/assets/982861d8-9a86-413d-8538-a4bf9fb818ee" />

    <img width="1212" height="952" alt="Image" src="https://github.com/user-attachments/assets/1516da57-02a0-4d42-bc99-6778618ec05d" />
    - Update Form

    <img width="543" height="944" alt="Image" src="https://github.com/user-attachments/assets/bf99eba6-1e5e-4c1a-8ac2-e16ac34ee9a8" />

    <img width="536" height="934" alt="Image" src="https://github.com/user-attachments/assets/577fc512-9865-4ded-aaf1-dc1632ba3c17" />
    - View Form

    <img width="1175" height="924" alt="Image" src="https://github.com/user-attachments/assets/3660cd46-14df-4917-906e-0565264324cc" />

    <img width="1209" height="939" alt="Image" src="https://github.com/user-attachments/assets/09581f99-b728-4975-9b94-f62f2a72507e" />

- Date picker looks good now

    <img width="507" height="439" alt="Image" src="https://github.com/user-attachments/assets/01579ff6-ee68-4108-a991-f73777d3d68e" />

- Materials Page

    <img width="1909" height="1034" alt="Image" src="https://github.com/user-attachments/assets/fe2be99d-e688-48c5-ab16-784fc64fcca5" />

    <img width="1910" height="1036" alt="Image" src="https://github.com/user-attachments/assets/58616159-11bd-4921-9b1a-b567b35730cb" />
    - View Usage Record

    <img width="1193" height="934" alt="Image" src="https://github.com/user-attachments/assets/711379c8-1b63-4b9a-894e-da5fc5ad021c" />

    <img width="1203" height="946" alt="Image" src="https://github.com/user-attachments/assets/3236ddff-176a-477e-a330-825511bf847a" />
    - View Updates 

    <img width="1211" height="924" alt="Image" src="https://github.com/user-attachments/assets/8b5e5a1d-f62f-41ef-9158-2e4a86f2a6fe" />

    <img width="1229" height="949" alt="Image" src="https://github.com/user-attachments/assets/ea433eef-4718-4eb8-b7d1-c083631f8fa1" />
 
    - Edit Updates

    <img width="537" height="934" alt="Image" src="https://github.com/user-attachments/assets/ade8099e-eb86-42da-a1ac-3e57fe1facf6" />

    <img width="594" height="942" alt="Image" src="https://github.com/user-attachments/assets/e09504dc-5e0c-41b6-a4f6-ae919ac185e7" />

## Issues
1. View Purchases does not have the color coded Status when in Dark Mode
- View Purchases

    <img width="1185" height="942" alt="Image" src="https://github.com/user-attachments/assets/5cfefbc1-d167-4f8b-a54e-11e9680a74e9" />

    <img width="1207" height="955" alt="Image" src="https://github.com/user-attachments/assets/c8adc661-49fd-4e64-9e46-7a2089ccfc1e" />
2.  Could you please clarify whether this is the intended behavior? In dark mode, there appears to be additional padding at the top and bottom in both pages, which causes the page layout to shift slightly. My understanding is that dark mode should primarily change the color scheme without affecting the layout or spacing.

    <img width="1915" height="1030" alt="Image" src="https://github.com/user-attachments/assets/d362e311-fa24-4eb1-980b-2c36d7579ae7" />

    <img width="1914" height="1035" alt="Image" src="https://github.com/user-attachments/assets/e507a79a-3aee-42bc-a115-d27a3de3b934" />

3. The area at the margins of the components/[age have inconsistent colors between the pages. Could this color just be the same as the main component background color. I know in Light Mode its one uniform white background. 
  - Material Page

    <img width="1910" height="1036" alt="Image" src="https://github.com/user-attachments/assets/3c170df0-1ba7-4098-bdea-c796e903d711" />
  - Consumables Page
    <img width="1917" height="1036" alt="Image" src="https://github.com/user-attachments/assets/5b1f1add-fd44-4d2a-9018-3ef458204477" />

    <img width="1915" height="1037" alt="Image" src="https://github.com/user-attachments/assets/95cc6c13-ccc8-4245-b7ac-a99f6c13fc5f" />

Note: Assuming the Add Material, Edit Name/Measurement, and View Update History buttons are non-functional

DeepighaJ reviewed (changes requested):
- Tested the PR, the model dark mode issues are fixed .
- Add Material, Edit Name/Measurement, View Update History, Edit icon on Usage Record are not clickable, assuming its intentional.

**Issue:**
- As mentioned previous review the color codes of Rejected/Approved in Purchase record model are not implemented in dark mode and need improvement.
- The inline error message and New available is not in Red in Update Consumable Form in dark mode.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/c1fe4586-a11e-491e-bb95-adb153bc33c4" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/fd0367eb-0c00-40db-954c-a712e3d22e55" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/5bd7c8d9-e300-45f4-8fc7-53766ec84cf9" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/2bdc2057-56e0-4e42-92e2-dc0cc3b5789d" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/7b36bbc0-ea50-409e-a266-575da8a3579a" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/22c89436-06db-44d3-8909-648055b51514" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/ff2830d2-06c4-4d6a-8466-25a9cb62a1fe" />

AmaanSyed09 committed:
Fix dark mode colors in Purchase Record and Update Consumable form

AmaanSyed09 committed:
Fix dark mode layout and page background consistency

DeepighaJ reviewed (changes requested):
Tested the PR locally.
The color coding issue is resolved in dark mode in Consumables page.
The inline error messages are in red as expected in Consumables page.

Issue:
Inline error message is not in Red in Materials page -> Update Material Form model.

Materials Page:
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/0ff2de3a-ffc7-4960-afc8-94662d77541e" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/61bf4b4f-1e21-4d12-b924-099001581d0e" />

Consumables:
<img width="1600" height="905" alt="Image" src="https://github.com/user-attachments/assets/873f2e20-82cf-40c0-802c-16dd9ce3e789" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/acfea7a8-d499-4c28-a02b-07fd35d4cc2d" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/c8de3dd6-cdb1-4b13-be10-d13ba0fc57e4" />

AmaanSyed09 committed:
Fix dark mode validation error colors in Update Material form

AmaanSyed09 commented:
Addressed the remaining issues.

Fixed the reviewer-reported inline validation color issue in the Materials page Update Material Form. The dark-mode global text styling was overriding the existing semantic error color and causing the “Quantity Used exceeds the available stock” and “Sum of Used and Wasted values exceeds available stock” messages to appear white instead of red. Added a localized dark-mode override for the Update Material form so these validation messages retain their intended red color without affecting other modal text or global styling.

Tested the Update Material Form in both light and dark mode and verified that the inline validation messages remain red in both themes while the existing form styling, calculations, and validation behavior remain unchanged.

Ready for re-review.

kzou55 reviewed (changes requested):
Hi Amaan, 

I ran and tested the PR locally again. 

## Verified
- No more layout shift due to extra vertical padding in dark mode
- Materials Page Improvements
  - Margin background color improved

    <img width="1909" height="990" alt="Image" src="https://github.com/user-attachments/assets/5c7c2db4-d258-4c7e-a874-98a8457c07d2" />

## Issues 
1. Consumables page's background color for its margins is inconsistent, think it be one uniform background like the materials page. (The area highlighted in orange in the image below) 

    <img width="1658" height="885" alt="Image" src="https://github.com/user-attachments/assets/07711a65-717f-418f-b92d-b0ebe4482ee0" />

    <img width="1663" height="885" alt="Image" src="https://github.com/user-attachments/assets/74ad9f5d-4b19-4316-9551-639e2bb1df0b" />

    <img width="1912" height="885" alt="Image" src="https://github.com/user-attachments/assets/38e7e634-4c10-4ebf-bf95-48230c16ac4b" />

2. Purchase Record Status Color Coded Text missing Color in Dark Mode for the status "Pending" for both Consumables/Materials Page

    <img width="1228" height="958" alt="Image" src="https://github.com/user-attachments/assets/4d68a47a-9a86-40b6-a603-9196bcc7a8a8" />

    <img width="1174" height="938" alt="Image" src="https://github.com/user-attachments/assets/d6e49755-87d0-475c-b766-30cc2a8c6c39" />

3. Text color for "New Available" number has inconsistent color. In consumable's page it has red when negative while white in the material's page

    <img width="586" height="953" alt="Image" src="https://github.com/user-attachments/assets/c97078c0-2f74-48bb-abc5-fcd514fd826d" />

    <img width="566" height="952" alt="Image" src="https://github.com/user-attachments/assets/293a13e0-3910-4c8b-8fc9-8472136abd07" />

AmaanSyed09 committed:
Fix remaining inventory dark mode and layout issues

AmaanSyed09 commented:
Hi @kzou55 
Regarding the Consumables outer background, the inventory pages currently follow an existing alternating background pattern. Materials, Equipment, and Tools use the standard outer background, while Consumables, Reusables, and Unit of Measurement use the darker outer background. A similar alternating pattern is also used for the inventory table rows. Because of this existing pattern, changing only Consumables to match Materials would make it inconsistent with Reusables and the Unit of Measurement. Making the background uniform across the related pages would broaden the scope beyond the specific Consumables change.

I can make the change if that is the intended design; it should not be an issue. I just wanted to confirm before modifying the existing pattern. Would you still like Consumables changed to match Materials, even though it would differ from the other related inventory pages?

AmaanSyed09 commented:
Addressed the remaining issues.

Fixed the Purchase Record status color coding so Pending now retains its orange/yellow semantic color in dark mode for both Materials and Consumables, while Approved remains green and Rejected remains red. Improved the shared Purchase Record layout so long Materials values no longer push the Status and Action columns out of view, kept Approve and Reject aligned side-by-side, and matched the dark-mode Action-column spacing with the existing light-mode layout. Aligned the Materials Usage Record, Updates, and Purchases View actions to match the existing Consumables table layout. Also fixed the Materials Update Material Form so a negative New Available value correctly uses the existing red semantic styling in dark mode, consistent with Update Consumable.

Tested the updated Purchase Record modal on both Materials and Consumables in light and dark mode, including status colors, long table content, Action-column layout, and Approve/Reject alignment. Verified the Materials table action alignment and confirmed the Update Material negative New Available value and inline validation messages display correctly in dark mode.

Ready for re-review.

kzou55 reviewed (dismissed):
Hi Amaan, 

I ran and tested the PR locally again. 

## Verified
- Improvements/Change made:
  - Purchase Record Component has the color coded status text

   <img width="1209" height="963" alt="Image" src="https://github.com/user-attachments/assets/91b63f92-f35b-48b6-85d4-137e65c3f90d" />
  - Update Form has consistent red text

    <img width="557" height="934" alt="Image" src="https://github.com/user-attachments/assets/843c04c4-6af5-4954-8b77-882b4ca7c253" />

    <img width="558" height="942" alt="Image" src="https://github.com/user-attachments/assets/2c186a81-1cc0-476d-b727-ffb6c6c50eb7" />

- Previous behavior working as intended

  https://github.com/user-attachments/assets/13aba1f0-4f98-4fa9-ac53-f48e33ff56ba

AmaanSyed09 commented:
Hi @DeepighaJ, kzou55 also tested the latest changes locally and confirmed that the inline validation errors on the Materials Update Form are displaying in red as expected. I tested the latest vamsidhar-consumables-dark-mode branch again locally as well, and the Quantity Used error, combined Used + Wasted error, and negative New Available value are all displaying in red in Dark Mode. The current implementation appears to be working as intended.

<img width="454" height="807" alt="image" src="https://github.com/user-attachments/assets/3e79eba1-bda1-4e5b-8121-4738b615a220" />

<img width="1420" height="780" alt="image" src="https://github.com/user-attachments/assets/319cceee-528e-4ad6-b62c-87c425a17431" />

<img width="1433" height="776" alt="image" src="https://github.com/user-attachments/assets/18f25413-ef04-41a2-947d-debd59b010e7" />

AmaanSyed09 commented:
Addressed the remaining issues.

Merged the latest development branch into PR #4402 and resolved the Item List conflicts while preserving the approved Consumables dark mode implementation, filter layout, table styling, low-stock indicators, and existing action behavior. Integrated the compatible Materials updates, including Usage %, Stock Health, and Material Usage chart functionality.

Verified the Materials and Consumables pages in light and dark mode, including filters, status color coding, edit forms, validation states, Purchase Record actions, and Material Usage Record functionality. All resolved files pass the applicable checks, and the updated branch has been pushed.

Ready for Merge.
